lib/PlansConfigurator/main.tsx

```diff
@@ -0,0 +1,40 @@
+import { createRoot } from 'react-dom/client';
+import PlansConfigurator from './PlansConfigurator';
+import { ISources } from './interface';
+
+declare global {
+    interface Window {
+        mobbexManual: boolean
+        mobbexSources: ISources,
+        mobbexFeaturedPlans: string[],
+        mobbexSelectedPlans: string[],
+        mobbexAdvancedPlans : string[],
+        mobbexShowFeaturedPlans: boolean,
+    }
+}
+
+const divId = "mbbx-plans-configurator";
+let root: ReturnType<typeof createRoot> | null = null;
+console.log("mobbex sources:", window.mobbexSources)
+
+function renderPlansConfigurator() {
+    const container = document.getElementById(divId);
+    if (container) {
+        if (root) {
+            root.unmount();
+        }
+        root = createRoot(container);
+        root.render(
+          <PlansConfigurator
+            manual={window.mobbexManual || false}
+            sources={window.mobbexSources || undefined}
+            featuredPlans={window.mobbexFeaturedPlans || []}
+            selectedPlans={window.mobbexSelectedPlans || []}
+            advancedPlans={window.mobbexAdvancedPlans || []}
+            showFeaturedPlans={window.mobbexShowFeaturedPlans || false}
+          />
+        );
+    }
+}
```

Comment:
Agrego comentarios, mensajes de error y cambio un poco el formato con clausulas de guarda. Por otro lado, muy bueno el manejo del type en root
```suggestion
let root: ReturnType<typeof createRoot> | null = null;

function renderPlansConfigurator() {
    const container = document.getElementById("mbbx-plans-configurator");

    // Exit on container not found
    if (!container)
        return console.error("Mobbex Plans Configurator container not found");

    // If it's already rendered, unmount it first
    if (root) root.unmount();

    root = createRoot(container);
    root.render(
      <PlansConfigurator
        manual={window.mobbexManual || false}
        sources={window.mobbexSources || undefined}
        featuredPlans={window.mobbexFeaturedPlans || []}
        selectedPlans={window.mobbexSelectedPlans || []}
        advancedPlans={window.mobbexAdvancedPlans || []}
        showFeaturedPlans={window.mobbexShowFeaturedPlans || false}
      />
    );
}
```